### PR TITLE
Patch 1 solution for "Replace toml package usage #133"

### PR DESCRIPTION
**Pull request by JawadHalabiMTB**
That's my solution for this issue please LMK about any missing changes or confusing

### CODE REVIEW

**Comment by virtuald** [comment]
Oh, I forgot that we use dumps. tomlkit is a better choice then.

**Comment by auscompgeek** [comment]
We could use tomli-w? Either way I'm not fussed.

File: robotpy_build/tool.py [hunk 1]
```diff
@@ -251,7 +251,7 @@ def add_subparser(cls, parent_parser, subparsers):
             help="Displays pyproject.toml after override processing",
             parents=[parent_parser],
         )
-        parser.add_argument("toml", nargs="?", default="pyproject.toml")
+        parser.add_argument("tomli", nargs="?", default="pyproject.toml")
```
Comment (auscompgeek) [inline robotpy_build/tool.py:254]: Please don't change this argument name. `tomli` as an argument name won't make much sense from a user perspective.

File: robotpy_build/overrides.py [hunk 2]
```diff
@@ -48,18 +48,18 @@ def apply_overrides(d: Dict, override_keys: List[str]):
...
+    import tomli
 
     parser = argparse.ArgumentParser()
     parser.add_argument("fname")
     parser.add_argument("key")
     args = parser.parse_args()
 
     with open(args.fname) as fp:
-        d = toml.load(fp)
+        d = toml.loads(fp)
```
Comment (TheTripleV) [inline robotpy_build/overrides.py:59]:
```suggestion
        d = tomli.loads(fp)
```

File: robotpy_build/tool.py [hunk 3]
```diff
@@ -265,11 +265,11 @@ def run(self, args):
         p = platforms.get_platform(args.platform)
         override_keys = platforms.get_platform_override_keys(p)
 
-        with open(args.toml) as fp:
-            d = toml.load(fp)
+        with open(args.tomli) as fp:
+            d = toml.loads(fp)
```
Comment (TheTripleV) [inline robotpy_build/tool.py:269]:
```suggestion
            d = tomli.loads(fp)
```

File: robotpy_build/tool.py [hunk 4]
```diff
@@ -314,7 +314,7 @@ def run(self, args):
             self.arch_names.add(plat.arch)
 
         with open(args.toml_link) as fp:
-            config = toml.load(fp)["tool"]["robotpy-build"]
+            config = toml.loads(fp)["tool"]["robotpy-build"]
```
Comment (TheTripleV) [inline robotpy_build/tool.py:317]:
```suggestion
            config = tomli.loads(fp)["tool"]["robotpy-build"]
```

**Comment by JawadHalabiMTB** [comment]
Then let's go with tomlkit : ) we don't use poetry don't we ?

**Review by JawadHalabiMTB — COMMENTED** [review]
Done with review

**Review by JawadHalabiMTB — COMMENTED** [review]
small fix on overrides.py

**Review by JawadHalabiMTB — COMMENTED** [review]
add dumps to print function on tool.py

File: robotpy_build/setup.py [hunk 5]
```diff
@@ -2,7 +2,8 @@
 from os.path import abspath, exists, join
 from setuptools import find_packages, setup as _setup
 from setuptools_scm import get_version
-import toml
+from tomlkit import parse
+from tomlkit import dumps
```
Comment (auscompgeek) [inline robotpy_build/setup.py:6]: I would prefer not using from import for (de)serialisation things.

Comment (JawadHalabiMTB) [reply robotpy_build/setup.py:6]: Thanks for your replays first , What is your concern ? am I going in the right direction ? i would be happy to have answers

Comment (JawadHalabiMTB) [reply robotpy_build/setup.py:6]: So I can call the function as follow : 
1.import tomlkit 
2. tomlkit.loads()
3. tomlkit.parse()

good enough ?

Comment (virtuald) [reply robotpy_build/setup.py:6]: Yes, that seems like the standard way to use those functions.

**Comment by virtuald** [comment]
Thanks!

**Pull request merged**